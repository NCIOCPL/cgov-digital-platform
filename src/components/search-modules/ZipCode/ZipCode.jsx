import React from 'react';
import {Fieldset, TextInput} from '../../atomic';

const ZipCode = ({handleUpdate}) => {
  return (
    <Fieldset
      id="zip"
      legend="U.S. Zip Code"
      helpUrl="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#basicsearch">
      <TextInput action={handleUpdate} name="zip" id="z" label="Show trials near this U.S. ZIP code." maxLength={5} />
    </Fieldset>
  );
};


export default ZipCode;
