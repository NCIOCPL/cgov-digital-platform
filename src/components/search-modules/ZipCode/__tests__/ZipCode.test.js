import React from 'react';
import { render, shallow, mount } from 'enzyme';
import {
  resetDom,
  setupDom
} from '../../../../utilities/testHelpers';
import ZipCode from '../ZipCode';

const defaultProps = {
}

const setup = (enzymeMethod = render, props = {}) => {
  const component = enzymeMethod(
    <ZipCode {...props} />
  );
  return component;
};

describe('ZipCode', () => {
  describe('Render', () => {
    let component = null;

    afterEach(() => {
      resetDom();
    });

    it('renders without error', () => {
      component = setup(render, defaultProps);
      expect(component).toMatchSnapshot();
    });
  });

});
