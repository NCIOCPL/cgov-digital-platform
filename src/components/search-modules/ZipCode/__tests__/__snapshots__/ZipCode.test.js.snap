// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZipCode Render renders without error 1`] = `
<fieldset
  class="cts-fieldset "
  id="fieldset--zip"
>
  <legend
    class="cts-fieldset__legend"
  >
    <span>
      U.S. Zip Code
    </span>
  </legend>
  <a
    aria-label="Help"
    class="text-icon-help"
    href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#basicsearch"
    rel="noopener noreferrer"
    target="_blank"
  >
    ?
  </a>
  <div
    class="cts-fieldset__body"
  >
    <div
      class=""
    >
      <label
        class="cts-label"
        for="z"
        id="z-label"
      />
      <input
        aria-labelledby="z-label"
        aria-required="false"
        class="cts-input null"
        id="z"
        maxlength="5"
        spellcheck="false"
        type="text"
        value=""
      />
    </div>
  </div>
</fieldset>
`;
